client/js/components/procedure/basicSettings/DpBasicSettings.vue

```diff
@@ -146,6 +163,22 @@ export default {
   },
 
   methods: {
+    createAddonPayload () {
+      return {
+        type: this.addonPayload.resourceType,
+        attributes: this.addonPayload.attributes,
+        relationships: this.addonPayload.url === 'api_resource_update' ? undefined : {
+          procedure: {
+            data: {
+              type: 'Procedure',
+              id: this.procedureId
+            }
+          }
+        },
+        ...(this.addonPayload.url === 'api_resource_update' ? { id: this.addonPayload.id } : {}),
```

Comment:
What do you think about using destructuring here?

```suggestion
      const { attributes, id, resourceType, url } = this.addonPayload
      
      return {
        type: resourceType,
        attributes,
        relationships: url === 'api_resource_update' ? undefined : {
          procedure: {
            data: {
              type: 'Procedure',
              id: this.procedureId
            }
          }
        },
        ...(url === 'api_resource_update' ? { id } : {}),
```